.github/workflows/crucible-ci.yaml

```diff
@@ -10,205 +10,12 @@ on:
-        controller_tag: crucible-${{ github.run_number }}
-    - name: skip crucible-ci->integration-tests
-      if: ${{ ! matrix.scenario.enabled }}
-      run: echo "crucible-ci->integration-tests not enabled"
-
-  crucible-ci-self-hosted-runners:
-    runs-on: [ self-hosted, cpu-partitioning, remotehost ]
-    needs:
-    - generate-job-matrix-parameters
-    - crucible-ci-build-controller
-    strategy:
-      fail-fast: false
-      matrix:
-        scenario: ${{ fromJSON(needs.generate-job-matrix-parameters.outputs.self_hosted_scenarios) }}
-        userenv: ${{ fromJSON(needs.generate-job-matrix-parameters.outputs.userenvs) }}
-        ci_controller: [ "${{ needs.generate-job-matrix-parameters.outputs.build_controller }}" ]
-    steps:
-    - name: Matrix Parameters => (${{ matrix.scenario.enabled }}, ${{ matrix.scenario.endpoint }}, ${{ matrix.scenario.benchmark }}, ${{ matrix.userenv }}, ${{ matrix.ci_controller }})
-      run: |
-        echo "enabled=${{ matrix.scenario.enabled }}"
-        echo "endpoint=${{ matrix.scenario.endpoint }}"
-        echo "benchmark=${{ matrix.scenario.benchmark }}"
-        echo "userenv=${{ matrix.userenv }}"
-        echo "ci_controller=${{ matrix.ci_controller }}"
-    - name: checkout crucible
-      uses: actions/checkout@v3
-      with:
-        path: crucible
-    - name: checkout crucible-ci
-      uses: actions/checkout@v3
-      with:
-        repository: perftool-incubator/crucible-ci
-        ref: main
-        path: crucible-ci
-    - name: import secret
-      env:
-        CLIENT_SERVER_REGISTRY_AUTH_SECRET: ${{ secrets.CRUCIBLE_CI_ENGINES_REGISTRY_AUTH }}
-      if: env.CLIENT_SERVER_REGISTRY_AUTH_SECRET != ''
-      run: sudo bash -c "echo \"$CLIENT_SERVER_REGISTRY_AUTH_SECRET\" > /root/crucible-ci-engines-token.json"
-    - name: run crucible-ci->integration-tests
-      if: ${{ matrix.scenario.enabled }}
-      uses: ./crucible-ci/.github/actions/integration-tests
-      with:
-        artifact_tag: "crucible__self-hosted-runners__${{ matrix.scenario.endpoint }}-${{ matrix.scenario.benchmark }}-${{ matrix.userenv }}-${{ matrix.ci_controller }}"
-        ci_endpoint: "${{ matrix.scenario.endpoint }}"
-        scenarios: "${{ matrix.scenario.benchmark }}"
-        userenvs: "${{ matrix.userenv }}"
-        ci_target: "crucible"
-        ci_target_dir: ${{ github.workspace }}/crucible
-        ci_controller: ${{ matrix.ci_controller }}
-        controller_tag: crucible-${{ github.run_number }}
-    - name: skip crucible-ci->integration-tests
-      if: ${{ ! matrix.scenario.enabled }}
-      run: echo "crucible-ci->integration-tests not enabled"
-
-  crucible-ci-complete:
-    runs-on: ubuntu-latest
-    needs:
-    - crucible-ci-github-runners
-    - crucible-ci-self-hosted-runners
-    steps:
-    - name: Confirm Success
-      run: echo "crucible-ci-complete"
+  call-core-crucible-ci:
+    uses: perftool-incubator/crucible-ci/.github/workflows/core-crucible-ci.yaml@main
```

Comment:
@main is really required ? can we use any branch ?

Reply:
Apparently there has to be *something* there...it can be a branch name, a version, or a specific commit hash that I know of.  When developing this stuff I was using the branch name that I was submitting my code to in the crucible-ci repo (`dev-kmr`) and once that code was merged up I removed the `@dev-kmr` thinking I didn't need it anymore but it complained that it required something so I put the `@main` there to always point to the HEAD of the crucible-ci repo.